scripts/convert-idl.js

```diff
@@ -0,0 +1,88 @@
+'use strict';
+
+const fs = require('fs');
+const path = require('path');
+const Transformer = require('webidl2js');
+
+const allProperties = require('../lib/allProperties.js');
+const allExtraProperties = require('../lib/allExtraProperties.js');
+const { cssPropertyToIDLAttribute } = require('../lib/parsers.js');
+
+const srcDir = path.resolve(__dirname, '../src');
+const implDir = path.resolve(__dirname, '../lib');
+const outputDir = implDir;
+
+const propertyNames = [
+  ...allProperties,
+  ...Array.from(allExtraProperties).filter(prop => {
+    return prop !== 'css-float' && !allProperties.has(prop);
+  }),
+].sort();
+
+const camelCasedAttributes = propertyNames.map(n => cssPropertyToIDLAttribute(n));
+const webkitCasedAttributes = propertyNames
+  .filter(n => n.startsWith('-webkit-'))
+  .map(n => cssPropertyToIDLAttribute(n, true));
+const dashedAttributes = propertyNames.filter(n => n.includes('-'));
+
+{
+  // TODO: This should be natively supported by WebIDL2JS's Transformer
+  // https://github.com/jsdom/webidl2js/issues/188
+  const genIDL = fs.createWriteStream(
+    path.resolve(__dirname, '../src/CSSStyleDeclaration-properties.webidl'),
+    {
+      encoding: 'utf-8',
+    }
+  );
+
+  genIDL.write(
+    `// autogenerated by scripts/convert-idl.js. do not edit! ${new Date().toISOString()}
+
+partial interface CSSStyleDeclaration {
+  // https://drafts.csswg.org/cssom/#dom-cssstyledeclaration-camel_cased_attribute
+`
+  );
+
+  for (const camelCasedAttribute of camelCasedAttributes) {
+    genIDL.write(`  [CEReactions] attribute [TreatNullAs=EmptyString] CSSOMString _${camelCasedAttribute};
+`);
+  }
+
+  genIDL.write(`
+  // https://drafts.csswg.org/cssom/#dom-cssstyledeclaration-webkit_cased_attribute
+`);
+
+  for (const webkitCasedAttribute of webkitCasedAttributes) {
+    genIDL.write(`  [CEReactions] attribute [TreatNullAs=EmptyString] CSSOMString _${webkitCasedAttribute};
+`);
+  }
+
+  genIDL.write(`
+  // https://drafts.csswg.org/cssom/#dom-cssstyledeclaration-dashed_attribute
+`);
+
+  for (const dashedAttribute of dashedAttributes) {
```

Comment:
I think it makes more sense do something like

```js
for (const property of propertyNames) {
  if (!property.includes('-')) continue;
  genIDL.write(…);
}
```

It fits better with what the spec says:

> For each CSS property _property_ that is a supported CSS property, except for properties that have no "`-`" (U+002D) in the property name, the following partial interface applies where _dashed attribute_ is _property_.

For WebKit above,

```js
for (const property of propertyNames) {
  if (!property.startsWith('-webkit-')) continue;
  const webkitCasedAttribute = cssPropertytoIDLAttribute(property, /* lowercaseFirst = */ true);
  genIDL.write(…);
}
```